buttplug/src/server/comm_managers/btleplug/btleplug_adapter_task.rs

```diff
@@ -54,41 +54,43 @@ impl BtleplugAdapterTask {
       );
       return;
     };
-    if let Some(name) = properties.local_name {
+
+    let device_name = if let Some(name) = properties.local_name {
+      name
+    } else {
+      String::new()
+    };
+
+    if (!device_name.is_empty() || !properties.services.is_empty()) && !tried_addresses.contains(peripheral_id) {
       let span = info_span!(
         "btleplug enumeration",
-        address = tracing::field::display(properties.address),
-        name = tracing::field::display(&name)
+        address = tracing::field::display(format!("{:?}", peripheral_id)),
+        name = tracing::field::display(&device_name)
       );
       let _enter = span.enter();
       // Names are the only way we really have to test devices
       // at the moment. Most devices don't send services on
       // advertisement.
-      if !name.is_empty() && !tried_addresses.contains(peripheral_id)
-      //&& !connected_addresses_handler.contains_key(&properties.address)
+      debug!("Found new bluetooth device: {} {:?}", device_name, peripheral_id);
+      tried_addresses.push(peripheral_id.clone());
```

Comment:
This worries me: advertisement types arrive in random order, so if we see a manufacturer or services advertisement packet before a name advertisement, we could block a device we would match when the rest of the advertisement data arrives.